### PR TITLE
[move] Add beginnings of a move development guide

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Adds the beginnings of a move development guide.  Starts with some information on design patterns, and testing.  Will need more information and to grow, but I realized if I do the whole thing at once, it'll grow too big.

This is mainly to focus around:
1. Design patterns
2. Framework integrations especially around testing
3. Debugging (in the future)
4. Setting up tooling (in the future)

### Checklist

- Do all Lints pass?
  - [] Have you ran `pnpm spellcheck`? (this is already broken some other places, we'll need to clean up a bit later)
  - [x] Have you ran `pnpm fmt`?
  - [x] Have you ran `pnpm lint`?

### CODE REVIEW

File: apps/docusaurus/docs/move/move-on-aptos/development/design-patterns.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      allowlist: vector[],
+      denylist: vector[],
+    })
+  }
+
+  public fun only_allowed(caller: &signer) {
+    let caller_address = signer::address_of(caller);
+
+    let acl = borrow_global<AccessControl>(@deployer_address);
+    assert!(vector::contains(acl.allowlist), E_NOT_ALLOWED);
```
Comment (banool) [inline apps/docusaurus/docs/move/move-on-aptos/development/design-patterns.md:117]: This is missing the second argument for vector::contains. Also same thing, let's use error::something(E_SOMETHING).

**Comment by hariria** [comment]
Originally I was thinking about splitting the docs up so it was per repo based (I.e. sdk docs would live in sdk, indexer docs live in indexer repo, etc) so we could get source code and docs in the same repo and it'd be easier to do code references. But there's obvious trade offs with that - it's much more complex to maintain, navigation is more challenging, etc. We may revisit in the future but for now I think this is unfortunately the way we have to do things.

File: apps/docusaurus/docs/move/move-on-aptos/development/design-patterns.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+module deployer_address::error_example {
+
+  use mint_address::minter;
+
+  /// Mint multiple NFTs up to the amount requested
+  /// This will never fail, even if there's none to mint
+  public fun mint_many(caller: &signer, amount: u64) {
+    for (i in 0..amount) {
+      // Mint as many as we can, if there are no more, quit
+      if (0 == minter::amount_left()) {
```
[the remark this answers is not in the document]
Comment (gregnazario) [reply apps/docusaurus/docs/move/move-on-aptos/development/design-patterns.md:68]: This is an old habit for me from C.  If you put it on the left and you accidentally do a single = it'll fail in compilation vs the other way.

**Comment by hariria** [comment]
@gregnazario is it okay to close this?

**Pull request closed without merging**